### PR TITLE
Extend swirl-tag

### PR DESCRIPTION
**Pull request by joseabernardes**
## Summary

- [Ticket](https://linear.app/flip/issue/BSW-34/extend-swirl-tags-component)
- [Design](https://www.figma.com/design/iRHiAFsHxnzux8gQ9EjlZe/Swirl-Components?node-id=152-4591&m=dev)
- [Storybook](#)

### General

- [x] [Changeset added](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md)
- [x] The submitted code is organized and formatted according to our [💅 Style Guide](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-style-guide--docs).
  - [ ] The code is formatted with Prettier.
  - [ ] There are no linting errors.

### For new or updated components

https://swirl-storybook.flip-app.dev/?path=/docs/contributions-merge-publish--page

- [x] The changes do not contain any breaking changes.
- [x] The changes do not introduce new components that don't belong in the library (e.g. non-reusable components, highly specialized components, components including business logic)
- [ ] The component documentation is updated.
- [ ] The changes meet the [🤖 testing requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-testing--docs).
  - [ ] New features are tested.
  - [ ] In case of bug fixes, regression tests have been added.
  - [ ] All tests are 🟢.
- [ ] The changes meet the [🧏‍♀️ accessibility requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-accessibility--docs).
  - [ ] WCAG 2.1 Level A and Level AA requirements are met.
  - [ ] The Storybook a11y addon shows no errors.
  - [ ] The changes have been tested with a screen reader.
  - [ ] Keyboard controls have been tested, if applicable.
  - [ ] Components implementing form controls (inputs, buttons, selects, etc.) do not use Shadow DOM, and instead use Stencil's scoping mechanism
- [ ] The changes use our [🌈 theming concept](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-theming--docs).
  - [ ] Design tokens have been used where appropriate.
  - [ ] The component has been visually checked in combination with the "Light" and "Dark" theme.
- [ ] The changes meet our [🌍 internationalization requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-internationalization--docs).
  - [ ] No static text is used.
  - [ ] The component doesn't break with longer texts or different text wrappings.
  - [ ] Number, currency and date values can be formatted appropriately.
- [ ] The changes work in all supported browsers and viewports. See [📱 Responsive Design](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-responsive-design--docs)

### CODE REVIEW

File: packages/swirl-components/src/components/swirl-tag/swirl-tag.tsx [hunk 1]
```diff
@@ -26,15 +30,42 @@ export type SwirlTagSize = "s" | "m";
...
+  }
+
+  private forceIconProps() {
+    const icon = this.iconEl?.children[0];
+
+    icon?.setAttribute("size", "16");
+  }
+
+  private forceVariant() {
+    if (Boolean(this.bordered)) {
```
Comment (joseabernardes) [inline packages/swirl-components/src/components/swirl-tag/swirl-tag.tsx:61]: To keep this change backwards compatible, I implemented this fallback of the `bordered` prop to the new outline `variant`.

File: packages/swirl-components/src/components/swirl-tag/swirl-tag.css [hunk 2]
```diff
@@ -29,62 +29,146 @@
...
+    color: var(--s-icon-warning);
   }
 
-  &.tag--bordered {
+  &.tag--variant-outline {
     border-color: var(--s-border-warning);
   }
+
+  &.tag--variant-strong {
+    color: var(--s-text-on-action-primary);
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-tag/swirl-tag.css:104]: Issue: Contrast is too low. I think we need dark text here.
Bildschirmfoto 2024-07-17 um 11 33 18

File: packages/swirl-components/src/components/swirl-tag/swirl-tag.css [hunk 3]
```diff
@@ -29,62 +29,146 @@
...
+  & .tag__removal-button {
+    color: var(--s-icon-on-image);
+  }
+
+  &.tag--variant-outline {
+    border-color: var(--s-text-on-image);
+  }
+}
+
+.tag__icon {
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-tag/swirl-tag.css:170]: Issue: The position is off. I think it's missing a `margin-left: calc(-1 * var(--s-space-4))`
Bildschirmfoto 2024-07-17 um 11 35 57
vs.
Bildschirmfoto 2024-07-17 um 11 35 53

**Review by Sqrrl — CHANGES_REQUESTED** [review]

File: packages/swirl-components/src/components/swirl-tag/swirl-tag.css (hunk 2 above)
Comment (joseabernardes) [reply packages/swirl-components/src/components/swirl-tag/swirl-tag.css]: Good catch, thanks.
Talked with the designers and they also agreed.

File: packages/swirl-components/src/components/swirl-tag/swirl-tag.css (hunk 3 above)
Comment (joseabernardes) [reply packages/swirl-components/src/components/swirl-tag/swirl-tag.css:170]: Right, I missed that.

**Review by Sqrrl — APPROVED** [review]

**Pull request merged**